### DISCUSSION TITLE
Selection dialog after swipe right a favorite route:

### DISCUSSION THREAD

**Discussion by DolomitesTom** [devemux86/cruiser#646] [Q&A] [answered]
I use V5 sometimes on my oldest device Galaxy S8+ with Android 10, only to see if there are some improvements which could let me forget the loss of 3D. It didn't happen yet, but why should it stop me from making sensible suggestions:

Long-press on a route in "myRouteCollection" shows the context menu for also loading a route. Swipe right does the same, quick loads a route. Maybe it makes sense to ask if the route or overlay or both shall be loaded. Maybe only an additional entry in the context menu makes sense, therefore I would propose an additional entry under "load route" for loading a route plus Overlay, maybe also an additional entry with "load overlay" only.

**Comment by devemux86** [comment]
The overlay of favorites / routes / tracks is already available as "Show on map".
It allows to turn on / off the visibility and select the zoom level of appearance.

<img src="https://github.com/user-attachments/assets/8a1a8828-8596-4847-b061-a258fdc506d0" width="50%"/><img src="https://github.com/user-attachments/assets/64e4b055-9154-4206-a53b-6c12249540fe" width="50%"/>

**Comment by DolomitesTom** [comment]
yes I know, but this has also disadvantages. When you have several Overlays active via this "show on map" feature and want to deactive the overlay, then you have to manually search for the right one in several folders. This is complicated. When you load it via quick load over the context menu as propsosed, you just have to press Route -- delete -- overlay, when you want to deactivate it again. The overlay itself has no context menu where you can just longpress and then tap on delete.

**Comment by bios4** [comment]
Adding a touch listener to each route/overlay shown on the map would be really a great improvement! 😍

I remember that this was already briefly discussed [here](https://github.com/devemux86/cruiser/discussions/389#discussioncomment-7925869).
Are there any plans for adding this in Cruiser 5?

**Reply by devemux86** [reply]
We will see, as touch listener on lines is expensive.
Currently it only exists on the calculated route.

**Comment by devemux86** [comment]
> We can think if the swipe right could show a selection dialog for these actions, as a faster method.

This will be available in the next update. Selection dialog after swipe right a favorite route:

<img src="https://github.com/user-attachments/assets/c1008bc2-1150-4dfa-96b9-03033cfa7bc1" width="50%"/>

**Reply by bios4** [reply]
> This will be available in the next update. Selection dialog after swipe right a favorite route

"Show on map" in that new dialog has the same meaning as "Load as overlay", correct?

Because when doing the opposite and select Routing > Delete from the main menu, the user is asked whether he wants to delete the "Route" or the "Overlay" (or both).

So maybe "Load as overlay" would be the better naming here?

**Reply by devemux86** [reply]
It is the same function as the existing menu "Show on map" in favorite routes (and all other favorite types).
So we should have the same name for the same function in favorites.

If the "Routing | Clear" menu is confusing, we can change it.

**Reply by bios4** [reply]
Fair point.

What happens when I swipe right on a route which is already loaded and/or is already shown on the map?

Does the same dialog pop up?
Or will I be able to "unload" the route and remove the overlay from the map then?

Especially the latter (remove the overlay of the selected route) would be a nice benefit, especially when a lot of routes are loaded as overlay.

**Reply by devemux86** [reply]
Also if we implement automatic storage of imported route's overlay in favorite tracks,
then the clear dialog will not need anymore the overlay check box:

- https://github.com/devemux86/cruiser/discussions/431

**Reply by devemux86** [reply]
For new questions please write new posts, as the notification only includes the first text.

---

> What happens when I swipe right on a route which is already loaded and/or is already shown on the map?

The dialog always appears, we can uncheck the "Show on map" to hide the route's track.

**Reply by bios4** [reply]
> The dialog always appears, we can uncheck the "Show on map" to hide the route's track.

👍

> the clear dialog will not need anymore the overlay check box

When a route is loaded (independent if from "My collection" or imported from an external file) and an overlay is shown for it, I'd appreciate to have the option to delete EITHER the route, OR its overlay (or both)!

There are cases where I display an overlay of a certain route while having another one loaded for navigation.
This is achieved by 1) importing "Route + Overlay" for the 1st route - 2) deleting "Route" only from "Routing > Delete" - 3) import "Route only" for the 2nd route.

In these cases I DO NOT WANT to store these routes to "My collection".

So: please do NOT remove the separate checkbox from the Routing > Delete menu!

**Reply by devemux86** [reply]
(Favorites is the only way to store and remember permanent content in the app)

I have not thought yet if / how it could be done, anyway we better discuss it in its thread:
- https://github.com/devemux86/cruiser/discussions/431

**Reply by DolomitesTom** [reply]
> It is the same function as the existing menu "Show on map" in favorite routes (and all other favorite types).

How then can this overlay be deleted again later without having to search for the right route name in the myRoutes?

**Reply by devemux86** [reply]
You have to find it in my routes.

**Comment by bios4** [comment]
>We will see, as touch listener on lines is expensive.

“Expensive” in what context is meant here?
Map performance?
Development/implementation effort?
...?

Touch listeners would be the perfect solution for what is discussed above.
Without them you can use at least “color coding” to assign the respective route on the map, if there are a lot of routes displayed.

**Reply by devemux86** [reply]
Expensive in calculations. It will make the touch gesture to be slow. It will need careful design.
e.g. there are 10-20 tracks with thousands of points and try to find what is getting the gesture.

---

Please create a new thread to propose and discuss a different suggestion.
Here it is about the swipe gesture.

**Comment by DolomitesTom** [comment]
Wouldn't it be easier in programming to also load the overlay in the same way as the route? So that you can just press the menu route -- delete and then decide what to delete, route plus overlay or only one of both. 
Because this was the intention of my request to not need to search later. Or do you need to temporarily create an external file to import this as route + overlay and this would be technically also not that good?

**Reply by devemux86** [reply]
For map libraries all lines are the same: series of connected points.
We'll see in the future what is possible and with good performance.

**Comment by DolomitesTom** [comment]
Works as expected 👍

**Comment by devemux86** [comment, marked as answer]
- Selection dialog after swipe right a favorite route is available in https://github.com/devemux86/cruiser/discussions/684
